mender-auth/cli/cli.cpp

```diff
@@ -38,6 +40,109 @@ namespace setup = mender::common::setup;
 
 namespace context = mender::auth::context;
 namespace ipc = mender::auth::ipc;
+namespace cli = mender::common::cli;
+
+const vector<cli::Option> opts_bootstrap_daemon {
+	cli::Option {
+		.long_option = "forcebootstrap",
+		.short_option = "F",
+		.description = "Force bootstrap",
+	},
+	cli::Option {
+		.long_option = "passphrase-file",
+		.description =
+			"Passphrase file for decrypting an encrypted private key. '-' loads passphrase from stdin",
+		.default_value = "''",
+	},
+};
+
+const cli::Command cmd_bootstrap {
+	.name = "bootstrap",
+	.description = "Perform bootstrap and exit",
+	.options = opts_bootstrap_daemon,
+};
+
+const cli::Command cmd_daemon {
+	.name = "daemon",
+	.description = "Start the client as a background service",
+	.options = opts_bootstrap_daemon,
+};
+
+const conf::Paths default_paths {};
+
+const cli::App cli_mender_auth = {
+	.name = "mender-auth",
+	.short_description = "manage and start Mender Auth",
+	.long_description =
+		R"(mender-auth integrates both the mender-auth daemon and commands for manually
+   performing tasks performed by the daemon (see list of COMMANDS below).
+
+Global flag remarks:
+   - Supported log levels incudes: 'debug', 'info', 'warning', 'error', 'panic' and
+     'fatal'.
+
+Environment variables:
+   - MENDER_CONF_DIR - configuration (default: )"
+		+ default_paths.GetPathConfDir() + R"().
+   - MENDER_DATA_DIR - identity, inventory and update modules (default: )"
+		+ default_paths.GetPathDataDir() + R"().
+   - MENDER_DATASTORE_DIR - runtime datastore (default: )"
+		+ default_paths.GetDataStore() + R"().)",
+	.version = string {MENDER_VERSION},
+	.commands =
+		{
+			cmd_bootstrap,
+			cmd_daemon,
+		},
+	.global_options =
+		{
+			cli::Option {
+				.long_option = "config",
+				.short_option = "c",
+				.description = "Configuration FILE path",
+				.default_value = default_paths.GetConfFile(),
+				.parameter = "FILE"},
+			cli::Option {
+				.long_option = "fallback-config",
+				.short_option = "b",
+				.description = "Fallback configuration FILE path",
+				.default_value = default_paths.GetFallbackConfFile(),
+				.parameter = "FILE"},
+			cli::Option {
+				.long_option = "data",
+				.short_option = "d",
+				.description = "Mender state data DIRECTORY path",
+				.default_value = default_paths.GetPathDataDir(),
+				.parameter = "DIR"},
+			cli::Option {
+				.long_option = "log-file",
+				.short_option = "L",
+				.description = "FILE to log to",
+				.parameter = "FILE"},
+			cli::Option {
+				.long_option = "log-level",
+				.short_option = "l",
+				.description = "Set logging level",
+				.default_value = "info",
+			},
+			// TODO: not implemented
+			cli::Option {
+				.long_option = "trusted-certs",
+				.short_option = "E",
+				.description = "Trusted server certificates FILE path",
+				.parameter = "FILE"},
+			// TODO: not implemented
```

Comment:
?

Reply:
On this commit I just marked everything that was not implemented and later commits implement or remove the flags.

Should I squash all in one commit?

Reply:
Up to you really. I think either way is fine